src/item.cpp

```diff
@@ -8415,10 +8416,16 @@ void item::set_snippet( const snippet_id &id )
     snip_id = id;
 }
 
-const item_category &item::get_category() const
+const item_category &item::get_category( bool insides ) const
 {
     static item_category null_category;
-    return type->category_force.is_valid() ? type->category_force.obj() : null_category;
+    if( insides && type->category_force == item_category_container &&
+        contents.num_item_stacks() == 1 ) {
+        const item &ci = contents.only_item();
+        return ci.type->category_force.is_valid() ? ci.type->category_force.obj() : null_category;
```

Comment:
```suggestion
        return ci.get_category();
```
Or possibly this, in case the contained item is a container as well (e.g. a bottle of something within a backpack).
```suggestion
        return ci.get_category( true );
```

Reply:
I have picked ci.get_category() for now. I don't know if multiple nested containers should be classified by the items that is store in the innermost container. Like, if we have 10 nested backpacks with a single aspirin pill - is that pile of stuff really drugs, or is it containers simply because there are more containers than actual contents?